jax/_src/xla_bridge.py

```diff
@@ -100,10 +101,11 @@
     help="Mock GPU client number of gpus.",
 )
 
-_CPU_ENABLE_GLOO_COLLECTIVES = config.DEFINE_bool(
```

Comment:
I'd leave this around for a few releases, but issue a DeprecationWarning if it is set.

Describe the changes in the CHANGELOG.md as well?